pkg/api/unversioned/time.go

```diff
@@ -79,6 +80,16 @@ func (t Time) Equal(u Time) bool {
 	return t.Time.Equal(u.Time)
 }
 
+// BeforeMicro reports whether the time instant t is before micro instance u.
+func (t Time) BeforeMicro(u v1.MicroTime) bool {
+	return t.Time.Before(u.Time)
```

Comment:
Shouldn't this round (er, truncate?) to nearest microsecond before comparing? Same for below function. If this function doesn't do anything special then maybe just delete it?